### PR TITLE
RIA-4748 Place all notifications to caseworkers behind a feature flag

### PR DESCRIPTION
**Pull request by burale**
### JIRA link (if applicable) ###

[RIA-4748](https://tools.hmcts.net/jira/browse/RIA-4748)

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/resources/scenarios/RIA-2976-aip-send-notifications-request-more-time-appellant-case-officer-clarifying-questions.json [hunk 1]
```diff
@@ -68,32 +69,18 @@
...
-          {
-            "id": "1020_SUBMIT_TIME_EXTENSION_CASE_OFFICER",
-            "value": "$/[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}/"
-          },
           {
             "id": "1020_MAKE_AN_APPLICATION_APPELLANT_AIP_EMAIL",
             "value": "$/[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}/"
           },
           {
-            "id": "1020_MAKE_AN_APPLICATION_APPELLANT_AIP_SMS",
```
Comment (bradsorour) [inline src/functionalTest/resources/scenarios/RIA-2976-aip-send-notifications-request-more-time-appellant-case-officer-clarifying-questions.json:80]: @burale is there any reason we're moving the AIP SMS notification from this test?

File: src/functionalTest/resources/scenarios/RIA-3025-paid-notification-legal-rep-after-appeal-submitted-after-listing.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "description": "RIA-3025 Send payment paid notification to legal representative (PA appeal type) after appeal submitted state (after listing)",
-  "enabled": false,
+  "disabled": true,
```
Comment (bradsorour) [inline src/functionalTest/resources/scenarios/RIA-3025-paid-notification-legal-rep-after-appeal-submitted-after-listing.json:3]: @burale what is the purpose of this change?

File: src/functionalTest/resources/scenarios/RIA-3082-pay-and-submit-notification.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
   "description": "RIA-3082 Send appeal submitted notification to Case Officer and Home Office after successful pay and submit",
-  "enabled": false,
+  "launchDarklyKey": "tcw-notifications-feature:true",
+  "disabled": true,
```
Comment (bradsorour) [inline src/functionalTest/resources/scenarios/RIA-3082-pay-and-submit-notification.json:4]: @burale we our key here if the test is disabled?

File: src/functionalTest/resources/scenarios/RIA-3313-appellant-ftpa-submitted-ho-notification-failure.json [hunk 4]
```diff
@@ -1,11 +1,13 @@
 {
   "description": "RIA-3313 appellant ftpa application submitted Home Office notification failure",
+  "launchDarklyKey": "tcw-notifications-feature:true",
   "request": {
     "uri": "/asylum/ccdAboutToSubmit",
     "credentials": "LegalRepresentative",
     "input": {
       "id": 3313,
       "eventId": "applyForFTPAAppellant",
+
```
Comment (bradsorour) [inline src/functionalTest/resources/scenarios/RIA-3313-appellant-ftpa-submitted-ho-notification-failure.json:10]: @burale can look to remove this whitespace

File: src/functionalTest/resources/scenarios/RIA-3631-pay-and-submit-notification.json [hunk 5]
```diff
@@ -1,6 +1,7 @@
 {
   "description": "RIA-3631 Send appeal submitted notification to Case Officer and exclude Home Office after successful pay and submit",
-  "enabled": false,
+  "launchDarklyKey": "tcw-notifications-feature:true",
+  "disabled": true,
```
Comment (bradsorour) [inline src/functionalTest/resources/scenarios/RIA-3631-pay-and-submit-notification.json:4]: @burale same as comments above

File: src/functionalTest/resources/scenarios/RIA-3631-send-appeal-submitted-and-paid-notification-legal-rep.json [hunk 6]
```diff
@@ -1,6 +1,7 @@
 {
   "description": "RIA-3631 Send appeal submitted and paid notification to legal representative (Home",
-  "enabled": false,
+  "launchDarklyKey": "tcw-notifications-feature:true",
+  "disabled": true,
```
Comment (bradsorour) [inline src/functionalTest/resources/scenarios/RIA-3631-send-appeal-submitted-and-paid-notification-legal-rep.json:4]: @burale same as comments above

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/caseofficer/CaseOfficerRespondentEvidenceSubmittedPersonalisation.java [hunk 7]
```diff
@@ -22,12 +23,14 @@
...
-        @NotNull(message = "respondentEvidenceSubmittedTemplateId cannot be null") @Value("${govnotify.template.respondentEvidenceSubmitted.caseOfficer.email}") String respondentEvidenceSubmittedTemplateId,
-        @Value("${iaExUiFrontendUrl}") String iaExUiFrontendUrl,
-        EmailAddressFinder emailAddressFinder
-    ) {
+            @NotNull(message = "respondentEvidenceSubmittedTemplateId cannot be null") @Value("${govnotify.template.respondentEvidenceSubmitted.caseOfficer.email}") String respondentEvidenceSubmittedTemplateId,
+            @Value("${iaExUiFrontendUrl}") String iaExUiFrontendUrl,
+            EmailAddressFinder emailAddressFinder,
+            FeatureToggler featureToggler) {
+        this.featureToggler = featureToggler;
         requireNonNull(iaExUiFrontendUrl, "iaExUiFrontendUrl must not be null");
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/caseofficer/CaseOfficerRespondentEvidenceSubmittedPersonalisation.java:34]: @burale look to run your formatted on this class `option + command + L`

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/caseofficer/CaseOfficerSubmitAppealPersonalisation.java [hunk 8]
```diff
@@ -21,12 +22,14 @@
...
     private final String iaExUiFrontendUrl;
     private final EmailAddressFinder emailAddressFinder;
+    private final FeatureToggler featureToggler;
 
     public CaseOfficerSubmitAppealPersonalisation(
-        @Value("${govnotify.template.appealSubmitted.caseOfficer.email}") String appealSubmittedCaseOfficerTemplateId,
-        @Value("${iaExUiFrontendUrl}") String iaExUiFrontendUrl,
-        EmailAddressFinder emailAddressFinder
-    ) {
+            @Value("${govnotify.template.appealSubmitted.caseOfficer.email}") String appealSubmittedCaseOfficerTemplateId,
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/caseofficer/CaseOfficerSubmitAppealPersonalisation.java:28]: @burale look to run your formatter on this class `option + command + L`

File: src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/caseofficer/CaseOfficerSubmitCasePersonalisation.java [hunk 9]
```diff
@@ -21,16 +22,18 @@
...
     private final String iaExUiFrontendUrl;
     private final EmailAddressFinder emailAddressFinder;
+    private final FeatureToggler featureToggler;
 
     public CaseOfficerSubmitCasePersonalisation(
-        @Value("${govnotify.template.submitCase.caseOfficer.email}") String submitCaseCaseOfficerTemplateId,
-        @Value("${iaExUiFrontendUrl}") String iaExUiFrontendUrl,
-        EmailAddressFinder emailAddressFinder
-    ) {
+            @Value("${govnotify.template.submitCase.caseOfficer.email}") String submitCaseCaseOfficerTemplateId,
```
Comment (bradsorour) [inline src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/caseofficer/CaseOfficerSubmitCasePersonalisation.java:28]: @burale look to run your formatter on this class `option + command + L`

Comment (bradsorour) [reply src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/caseofficer/CaseOfficerSubmitCasePersonalisation.java:28]: Please check formatting on all the classes below too.

File: src/main/resources/application.yaml [hunk 10]
```diff
@@ -973,9 +973,9 @@ feesAdminOfficerEmailAddress: ${FEES_ADMIN_OFFICER_EMAIL}
 paymentExceptionsAdminOfficerEmailAddress: ${PAYMENT_EXCEPTIONS_ADMIN_OFFICER_EMAIL:payment-exceptions-ao-aat@example.com}
 apcHomeOfficeEmailAddress: ${HOME_OFFICE_EMAIL_APC}
 lartHomeOfficeEmailAddress: ${HOME_OFFICE_EMAIL_LART}
-upperTribunalNoticesEmailAddress: ${IA_UPPER_TRIBUNAL_NOTICES_EMAIL}
-nbcEmailAddress: ${IA_NBC_EMAIL}
-ctscEmailAddress: ${IA_CTSC_EMAIL}
+upperTribunalNoticesEmailAddress: ${IA_UPPER_TRIBUNAL_NOTICES_EMAIL:upper-tribunal-notices@example.com}
```
Comment (bradsorour) [inline src/main/resources/application.yaml:976]: @burale we are no longer allowed to have defaults in this yaml file.

File: src/main/resources/application.yaml [hunk 11]
```diff
@@ -973,9 +973,9 @@ feesAdminOfficerEmailAddress: ${FEES_ADMIN_OFFICER_EMAIL}
...
+nbcEmailAddress: ${IA_NBC_EMAIL:respondent@example.com}
+ctscEmailAddress: ${IA_CTSC_EMAIL:respondent@example.com}
```
Comment (bradsorour) [inline src/main/resources/application.yaml:978]: @burale we are no longer allowed to have defaults in this yaml file.

File: src/functionalTest/java/uk/gov/hmcts/reform/iacasenotificationsapi/util/MapValueExtractor.java [hunk 12]
```diff
@@ -27,15 +36,94 @@ private MapValueExtractor() {
                 return null;
             }
 
+            Map map1 = (Map) value;
+
+            if (featureFlag == false) {
+
+                if (map1.containsKey("body")) {
```
Comment (bradsorour) [inline src/functionalTest/java/uk/gov/hmcts/reform/iacasenotificationsapi/util/MapValueExtractor.java:43]: @burale lets change this from `map1` to `expectationMap`

File: src/functionalTest/resources/scenarios/RIA-3631-send-appeal-submitted-and-paid-notification-legal-rep.json (hunk 6 above)
Comment (burale) [reply src/functionalTest/resources/scenarios/RIA-3631-send-appeal-submitted-and-paid-notification-legal-rep.json:4]: "disabled" makes more sense

File: src/main/resources/application.yaml (hunk 10 above)
Comment (burale) [reply src/main/resources/application.yaml]: Removed

File: src/functionalTest/java/uk/gov/hmcts/reform/iacasenotificationsapi/util/MapValueExtractor.java (hunk 12 above)
Comment (burale) [reply src/functionalTest/java/uk/gov/hmcts/reform/iacasenotificationsapi/util/MapValueExtractor.java]: Done

File: src/functionalTest/resources/scenarios/RIA-2976-aip-send-notifications-request-more-time-appellant-case-officer-clarifying-questions.json (hunk 1 above)
Comment (burale) [reply src/functionalTest/resources/scenarios/RIA-2976-aip-send-notifications-request-more-time-appellant-case-officer-clarifying-questions.json]: Added back

File: src/main/resources/application.yaml (hunk 11 above)
Comment (burale) [reply src/main/resources/application.yaml]: Removed

File: src/functionalTest/java/uk/gov/hmcts/reform/iacasenotificationsapi/CcdScenarioRunnerTest.java [hunk 13]
```diff
@@ -120,53 +138,66 @@ public void scenarios_should_behave_as_specified() throws IOException {
...
+                            .then()
+                            .log().ifError()
+                            .log().ifValidationFails()
+                            .statusCode(expectedStatus)
+                            .and()
+                            .extract()
+                            .body()
+                            .asString();
+
+            System.out.println("ACTUAL RESPONSE: \n" + actualResponseBody);
```
Comment (bradsorour) [inline src/functionalTest/java/uk/gov/hmcts/reform/iacasenotificationsapi/CcdScenarioRunnerTest.java:181]: @burale we should probably look to remove this output now.

File: src/functionalTest/java/uk/gov/hmcts/reform/iacasenotificationsapi/CcdScenarioRunnerTest.java [hunk 14]
```diff
@@ -120,53 +138,66 @@ public void scenarios_should_behave_as_specified() throws IOException {
...
+
+            System.out.println("EXPECTED RESULT: \n " + MapValueExtractor.extract(scenario, "expectation"));
```
Comment (bradsorour) [inline src/functionalTest/java/uk/gov/hmcts/reform/iacasenotificationsapi/CcdScenarioRunnerTest.java:183]: Same as above

File: src/functionalTest/resources/scenarios/RIA-1024-case-submitted-notification-to-case-officer-bradford-save-and-continue-enabled.json [hunk 15]
```diff
@@ -1,5 +1,6 @@
 {
-  "description": "RIA-1024 Send case submitted notification to case officer - bradford hearing centre (save and continue enabled)",
+  "description": "RIA-1024 KKK Send case submitted notification to case officer - bradford hearing centre (save and continue enabled)",
```
Comment (bradsorour) [inline src/functionalTest/resources/scenarios/RIA-1024-case-submitted-notification-to-case-officer-bradford-save-and-continue-enabled.json:2]: @burale the KKK text in the description should be removed.

File: src/functionalTest/resources/scenarios/RIA-2047-RIA-2694-RIA-3593-direction-request-hearing-requirements.json [hunk 16]
```diff
@@ -1,5 +1,7 @@
 {
   "description": "RIA-2047-RIA-2694-RIA-3593 direction to legal rep to submit hearing requirements",
+  "launchDarklyKey": "tcw-notifications-feature:true",
+  "id": "3631678_CASE_OFFICER_ADJOURN_HEARING_WITHOUT_DATE",
```
Comment (bradsorour) [inline src/functionalTest/resources/scenarios/RIA-2047-RIA-2694-RIA-3593-direction-request-hearing-requirements.json:4]: @burale is this id field required?

Comment (burale) [reply src/functionalTest/resources/scenarios/RIA-2047-RIA-2694-RIA-3593-direction-request-hearing-requirements.json:4]: Removed

File: src/functionalTest/resources/scenarios/RIA-1024-case-submitted-notification-to-case-officer-bradford-save-and-continue-enabled.json (hunk 15 above)
Comment (burale) [reply src/functionalTest/resources/scenarios/RIA-1024-case-submitted-notification-to-case-officer-bradford-save-and-continue-enabled.json:2]: Removed

File: src/functionalTest/java/uk/gov/hmcts/reform/iacasenotificationsapi/CcdScenarioRunnerTest.java (hunk 14 above)
Comment (burale) [reply src/functionalTest/java/uk/gov/hmcts/reform/iacasenotificationsapi/CcdScenarioRunnerTest.java:183]: Removed

File: src/functionalTest/java/uk/gov/hmcts/reform/iacasenotificationsapi/CcdScenarioRunnerTest.java (hunk 13 above)
Comment (burale) [reply src/functionalTest/java/uk/gov/hmcts/reform/iacasenotificationsapi/CcdScenarioRunnerTest.java:181]: Removed

**Review by bradsorour — DISMISSED** [review]
LGTM

**Review by bradsorour — DISMISSED** [review]
LGTM

**Review by bradsorour — APPROVED** [review]
LGTM
